src/openrct2-ui/interface/Window.cpp

```diff
@@ -1002,25 +975,40 @@ namespace OpenRCT2::Ui::Windows
         });
     }
 
-    bool WindowSetResize(WindowBase& w, const ScreenSize minSize, const ScreenSize maxSize)
+    bool WindowSetResize(WindowBase& w, ScreenSize minSize, ScreenSize maxSize)
     {
-        w.min_width = minSize.width;
-        w.min_height = minSize.height;
-        w.max_width = maxSize.width;
-        w.max_height = maxSize.height;
+        w.min_width = std::min(minSize.width, maxSize.width);
+        w.min_height = std::min(minSize.height, maxSize.height);
+        w.max_width = std::max(minSize.width, maxSize.width);
+        w.max_height = std::max(minSize.height, maxSize.height);
+
+        if (Config::Get().interface.EnlargedUi)
+        {
+            // Not sure why plugin windows have to be treated differently,
+            // but they currently show a deviation if we don't.
+            if (w.classification == WindowClass::Custom)
+            {
+                w.min_height += w.getTitleBarDiffTarget();
+                w.max_height += w.getTitleBarDiffTarget();
+            }
+            else
+            {
+                w.min_height += w.getTitleBarDiffNormal();
+                w.max_height += w.getTitleBarDiffNormal();
+            }
```

Comment:
This distinction has me stumped. If leave just the `else` block, however, plugin windows will look off. Conversely, if I leave just the other block, in-game windows like Park and NewRide will look slightly too short.